alembic/autogenerate/api.py

```diff
@@ -181,6 +181,41 @@ def _render_migration_diffs(context, template_args):
     )
 
 
+def generate_revision_from_script(config, migration_script):
+    from ..script import ScriptDirectory
+    script_directory = ScriptDirectory.from_config(config)
+
+    command_args = dict(
+        message=migration_script.message,
+        autogenerate=False,
+        sql=False,
+        head=migration_script.head or "head",
+        splice=migration_script.splice,
+        branch_label=migration_script.branch_label,
+        version_path=migration_script.version_path,
+        rev_id=util.rev_id(),
+        depends_on=migration_script.depends_on
+    )
+
+    revision_context = RevisionContext(
+        config, script_directory, command_args)
+
+    revision_context._last_autogen_context = AutogenContext(
+        None, opts={
+            'sqlalchemy_module_prefix': 'sa.',
+            'alembic_module_prefix': 'op.',
+            'user_module_prefix': None,
+        }, autogenerate=False)
+
+    revision_context.generated_revisions[0].upgrade_ops = migration_script.upgrade_ops
```

Comment:
can we say:

    revision_context.generated_revisions[0] = migration_script

?